ngraph/core/src/op/proposal.cpp

```diff
@@ -126,17 +160,9 @@ void op::v4::Proposal::validate_and_infer_types()
 {
     NGRAPH_OP_SCOPE(v4_Proposal_validate_and_infer_types);
     v0::Proposal::validate_and_infer_types();
-
-    const auto& class_probs_pshape = get_input_partial_shape(0);
-    const auto& class_bbox_deltas_pshape = get_input_partial_shape(1);
-    const auto& image_shape_pshape = get_input_partial_shape(2);
-    auto batch_size = class_probs_pshape[0];
-    if (class_probs_pshape.is_static() && class_bbox_deltas_pshape.is_static() &&
-        image_shape_pshape.is_static())
-        set_output_type(
-            1, get_input_element_type(0), PartialShape{batch_size * m_attrs.post_nms_topn});
-    else
-        set_output_type(1, get_input_element_type(0), PartialShape::dynamic());
+    // Output shape was inferred in v0's validate_and_infer_types
+    const auto out_ps = get_output_partial_shape(0);
+    set_output_type(1, get_input_element_type(0), PartialShape{out_ps[0]});
```

Comment:
We should check that the `out_ps` rank is not dynamic before accession `out_ps[0]`. If the rank is static then the output shape should be  PartialShape{Dynamic}

Reply:
Done